runtime/commonMain/src/kotlinx/serialization/SerialKinds.kt

```diff
@@ -26,6 +26,40 @@ import kotlinx.serialization.modules.*
  * represents a structure with two primitive fields.
  */
 public sealed class SerialKind {
+
+    /**
+     * Structure kind for singleton objects defined with `object` keyword.
+     * By default, objects are serialized as empty structures without any state and their identity is preserved
+     * across serialization within the same process, so you always have the same instance of the object.
+     *
+     * Empty structure is represented as a call to [Encoder.beginStructure] with the following [CompositeEncoder.endStructure]
+     * without any intermediate encodings.
+     */
+    public object OBJECT : SerialKind()
```

Comment:
AFAIR, we shouldn't have moved `OBJECT` out of structure kinds. The '*MOVE* objects to SerialKind' line in quip was referring to ENUM and CONTEXTUAL `object`s.

Reply:
Goot catch, thanks